src/graphics/program-lib/chunks/TBNderivative.frag

```diff
@@ -15,6 +15,7 @@ void getTBN() {
     vec3 B = dp2perp * duv1.y + dp1perp * duv2.y;
 
     // construct a scale-invariant frame
-    float invmax = 1.0 / sqrt( max( dot(T,T), dot(B,B) ) );
+    float denom = max( dot(T,T), dot(B,B) );
+    float invmax = (denom == 0.0) ? 0.0 : 1.0 / sqrt( denom );
```

Comment:
Does adding an 'if statement' affect performance in significant way here?

Reply:
I wouldn't expect a measurable difference. Check [this](https://github.com/playcanvas/engine/pull/3293#discussion_r663878611) answer.